### ISSUE TITLE
Type restrictions on list elements

### ISSUE THREAD

**Issue by glatard** [boutiques/boutiques#93]
See issue encountered in https://github.com/boutiques/boutiques/issues/92

We may be able to do this in the existing schema by just specifying that, when the input is a list, restrictions specified for the input apply to all the elements in the list. Thoughts?

**Comment by ttaa9** [comment]
Absolutely. That's what most people would expect to happen, in my opinion anyway. In fact, looking at LocalExec.py, the input validation code already assumes this (possibly not in line with the schema), if I am understanding you correctly.

**Comment by glatard** [comment]
Decision: all the restrictions on the  input will apply to all the list elements.
